### PR TITLE
Test fix: test slicing behaviour for parallel collection

### PR DESCRIPTION
**Pull request by JVerwolf**
Fixes testSupportsParallelCollectionAffectsSlicing so that it tests slicing behaviour.

### CODE REVIEW

**Comment by JVerwolf** [comment]
I ran this with` -Dtests.iters=100` a few times and it's all green.

File: server/src/test/java/org/elasticsearch/search/SearchServiceTests.java [hunk 1]
```diff
@@ -2040,42 +2043,133 @@ public void testSupportsParallelCollectionAffectsSlicing() throws Exception {
...
                 SearchContext searchContext = service.createContext(readerContext, request, task, ResultsType.NONE, true);
                 ContextIndexSearcher searcher = searchContext.searcher();
                 assertNotNull(searcher.getExecutor());
+                final long priorExecutorTaskCount = executor.getCompletedTaskCount();
                 searcher.search(termQuery, new TotalHitCountCollectorManager());
-                int expectedTaskCount = ++taskCount;
-                assertBusy(() -> assertEquals(expectedTaskCount, executor.getCompletedTaskCount()));
+                assertBusy(
+                    () -> assertEquals(
+                        "The number of slices should be 1 as NONE is does not support parallel collection.",
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/search/SearchServiceTests.java:2109]: remove "is"

**Review by javanna — APPROVED** [review]
LGTM

**Comment by JVerwolf** [comment]
I missed cleaning up the commit message on the squashed merge commit, and was checking if there was any recourse to fix it with the `revert` button, hence the autogenerated reference to a revert commit above ☝️. Oh well.

**Pull request merged**